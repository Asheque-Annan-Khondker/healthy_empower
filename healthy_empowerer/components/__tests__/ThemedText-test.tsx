import * as React from 'react';
import renderer from 'react-test-renderer';

import { ThemedText } from '../ThemedText';

it(`renders correctly`, () => {
  const tree = renderer.create(<ThemedText>Snapshot test!</ThemedText>).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly with styling props', () => {
    const tree = renderer.create(<ThemedText style={{ fontSize: 20, color: 'blue' }}>Styled text</ThemedText>).toJSON();
    expect(tree).toMatchSnapshot();
  });
 it('renders correctly with nested elements', () => {
    const tree = renderer.create(
      <ThemedText>
        <span>Nested Element</span>
      </ThemedText>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  }); 
  