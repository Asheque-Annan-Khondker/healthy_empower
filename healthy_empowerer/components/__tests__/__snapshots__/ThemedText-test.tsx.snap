// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#11181C",
      },
      {
        "fontSize": 16,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Snapshot test!
</Text>
`;

exports[`renders correctly with different text 1`] = `
<Text
  style={
    [
      {
        "color": "#11181C",
      },
      {
        "fontSize": 16,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Another test case
</Text>
`;

exports[`renders correctly with nested elements 1`] = `
<Text
  style={
    [
      {
        "color": "#11181C",
      },
      {
        "fontSize": 16,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <span>
    Nested Element
  </span>
</Text>
`;

exports[`renders correctly with styling props 1`] = `
<Text
  style={
    [
      {
        "color": "#11181C",
      },
      {
        "fontSize": 16,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      {
        "color": "blue",
        "fontSize": 20,
      },
    ]
  }
>
  Styled text
</Text>
`;
